fpc_imp_loire_tone.c

```diff
@@ -299,6 +301,10 @@ int64_t fpc_load_db_id(fpc_imp_data_t *data)
     ALOGD(__func__);
     fpc_data_t *ldata = (fpc_data_t*)data;
 
+    if(auth_id) {
```

Comment:
afaik this value *must* change whenever the fpc database changes, e.g. a new fingerprint is added or deleted.

Reply:
Tested that,can still add and remove fingerprints just fine.

Reply:
I never claimed adding/removing fingerprints wouldn't work.
But apps won't be able to tell they changed and invalidate their auth.

Reply:
Ah, then i misunderstood
So the caching needs to be invalidated upon enrolling or removing fingerprints. Is that correct?

Reply:
Jmm, this doesn't seem to be the case.
Tested app authentication (fingerprint gestures app) with a newly added fingerprint. This lead to successful authentication.
Tested app authentication with a deleted fingerprint. This lead to failing authentication.
Looks like the handle remains the same in all cases.

Reply:
nevermind the last statement, just checked without the pr, and added some logging for auth_id.
It does change when enrolling new fingerprints, and when removing all fingerprints.
So basically, as long as at least one fingerprint remains, and no new ones are enrolled, the auth_id remains the same.